src/ctia/domain/entities/campaign.clj

```diff
@@ -0,0 +1,13 @@
+(ns ctia.domain.entities.campaign
+  (:require
+    [ctia.properties :refer [get-http-show]]
+    [ctim.domain.id :as id]))
+
+(def short-id->long-id
+  (id/factory:short-id->long-id :campaign get-http-show))
+
+(defn with-long-id [{id :id, :as entity}]
+  (assoc entity :id (short-id->long-id id)))
+
+(defn page-with-long-id [{data :data, :as m}]
+  (assoc m :data (map with-long-id data)))
```

Comment:
I agree update is nicer than update-in.  If it will become more readable, I will change assoc to update.